apps/src/tutorialExplorer/tutorialExplorer.js

```diff
@@ -54,29 +66,205 @@ const TutorialExplorer = React.createClass({
       const itemIndex = this.state.filters[filterGroup].indexOf(filterEntry);
       newState = state.updateIn(['filters', filterGroup], arr => arr.splice(itemIndex, 1));
     }
-    this.setState(newState.toJS());
+
+    this.setState(newState.toJS(), function () {
+      const { filteredTutorials, filteredTutorialsForLocale } = this.filterTutorials(this.state.filters);
+      this.setState({
+        filteredTutorials: filteredTutorials,
+        filteredTutorialsCount: filteredTutorials.length,
+        filteredTutorialsForLocale: filteredTutorialsForLocale
+      });
+    });
```

Comment:
newState here is a set of changes to be applied to the state, but not the resulting state.  We need the resulting state to pass into filterTutorials.  And setState only creates a pending state transition, hence the use of the callback, so that we can get the resulting state.